easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1252,6 +1252,8 @@ def process_easyconfig(path, build_specs=None, validate=True, parse_only=False,
                 dep = ec.toolchain.as_dict()
                 _log.debug("Adding toolchain %s as dependency for app %s." % (dep, name))
                 easyconfig['dependencies'].append(dep)
+                if dep['hidden']:
+                    easyconfig['hiddendependencies'].append(dep)
```

Comment:
@geimer: I don't think this is required? I even think we should *not* do this, since it may have unwanted side-effects...

I'm working on another PR to your branch, where this is removed, but I figured I should double-check with you first why this was added.